src/main/java/com/haxwell/apps/quizki/entities/Reference.java

```diff
@@ -32,7 +38,20 @@ public void setText(String str) {
     	this.text = str;
     }
     
-    public String toString() {
+    @ManyToMany(fetch = FetchType.LAZY,
```

Comment:
Is there a reason this ManyToMany is defined differently than, say, the one in Choice?

Reply:
Yes, References are Many to Many so they have a Set of Questions. This way given a Reference you could get a set of all the Questions that use it. Same for Topics which also now have a set of Questions. Choices are only associated with a single Question not a set of them. Choices use a Join Column to hold the Foreign Key to the individual Question it's connected to, but that doesn't work when there are multiple Questions like here. So there is a Join Table used instead and that's the difference.

Reply:
Thank you.. and now that I go back and look, it is apparent, I mis-looked when I wrote this comment.. There actually is not a ManyToMany in Choice; it is a ManyToOne (as it should be).. and the other ManyToMany's are written this same way.. My bad, it was early.